src/synthesizer/line.py

```diff
@@ -364,34 +403,37 @@ def __next__(self):
             # Return the filter
             return self.lines[self.line_ids[self._current_ind - 1]]
 
-    def get_ratio_(self, ab):
+    def _get_ratio(self, line1, line2):
         """
-        Measure (and return) a line ratio
+        Measure (and return) a line ratio.
 
-        Arguments:
-            ab
-                a list of lists of lines, e.g. [[l1,l2], [l3]]
+        Args:
+            line1 (str)
+                The line or lines in the numerator.
+            line2 (str)
+                The line or lines in the denominator.
 
         Returns:
             float
                 a line ratio
         """
+        # If either line is a combination of lines check if we need to split
+        if line1 in self.lines:
+            line1 = [line1]
+        else:
+            line1 = [li.strip() for li in line1.split(",")]
+        if line2 in self.lines:
+            line2 = [line2]
+        else:
+            line2 = [li.strip() for li in line2.split(",")]
 
-        a, b = ab
-
-        # if a single value is given convert this into a list
-        if isinstance(a, str):
-            a = [a]
-        if isinstance(b, str):
-            b = [b]
-
-        return np.sum([self.lines[_line].luminosity for _line in a]) / np.sum(
-            [self.lines[_line].luminosity for _line in b]
-        )
+        return np.sum(
+            [self.lines[_line].luminosity for _line in line1]
+        ) / np.sum([self.lines[_line].luminosity for _line in line2])
```

Comment:
At the moment this sums over all axes right? So if we have an array (i.e. for different particles) they are automatically summed. I think we want this to be include axis=1 (or is it axis=0).

Reply:
Good catch! Fixed now.